### PR TITLE
fix(alerts/reports): ensure correct type is sent to api

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

- fixes issue where incorrect type is sent to the api, causing some validations to fail
- change action tooltips: Edit Alert -> Edit, Delete Alert -> Delete
- fix ts-ignore comment in AlertReportCronScheduler component 
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

- N/A
### TEST PLAN

- manual test:
  - Report/Alert Creation is successful
  - Report/Alert Edit is successful
  - Action tooltip copy looks correct
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx [hunk 1]
```diff
@@ -554,6 +554,7 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
 
     const data: any = {
       ...currentAlert,
+      type: isReport ? 'Report' : 'Alert',
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx:557]: ensures correct type is always sent to api. There are various code paths updating currentAlert, It's hard to tell which is causing this value to go out of sync with the UI.

File: superset-frontend/src/views/CRUD/alert/AlertList.tsx [hunk 2]
```diff
@@ -183,10 +183,6 @@ function AlertList({
     }
   };
 
-  useEffect(() => {
-    refreshData();
-  }, [isReportEnabled]);
-
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/alert/AlertList.tsx:189]: This was causing duplicate api calls when switching back and forth between alerts/reports. We no longer need this as baseFilters is changing which results in a new fetchData, which triggers a new api call.

File: superset-frontend/src/views/CRUD/hooks.ts [hunk 3]
```diff
@@ -206,119 +206,128 @@ export function useSingleViewResource<D extends object = any>(
...
+            updateState({
+              error: errMsg,
+            });
+          }),
+        )
+        .finally(() => {
+          updateState({ loading: false });
+        });
+    },
+    [handleErrorMsg, resourceName, resourceLabel],
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/hooks.ts:286]: noticed a few cases where errors were not resulting in toasts (after switching between alerts/reports and opening the modals various times). This fixes the error by ensuring we always have a correct reference to the current toast function.

File: superset-frontend/src/views/CRUD/hooks.ts [hunk 4]
```diff
@@ -206,119 +206,128 @@ export function useSingleViewResource<D extends object = any>(
...
-      });
-  }, []);
+            return errMsg;
+          }),
+        )
+        .finally(() => {
+          updateState({ loading: false });
+        });
+    },
+    [handleErrorMsg, resourceName, resourceLabel],
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/hooks.ts:329]: same as comment above

File: superset-frontend/src/views/CRUD/hooks.ts [hunk 5]
```diff
@@ -206,119 +206,128 @@ export function useSingleViewResource<D extends object = any>(
...
+            updateState({
+              error: errMsg,
+            });
+          }),
+        )
+        .finally(() => {
+          updateState({ loading: false });
+        });
+    },
+    [handleErrorMsg, resourceName, resourceLabel],
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/hooks.ts:245]: noticed a few cases where errors were not resulting in toasts (after switching between alerts/reports and opening the modals various times). This fixes the error by ensuring we always have a correct reference to the current toast function. 

This is the only change in this block

File: superset-frontend/src/views/CRUD/alert/AlertList.tsx (hunk 2 above)
Comment (dpgaspar) [reply superset-frontend/src/views/CRUD/alert/AlertList.tsx:189]: nice catch

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**